game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProTerritoryValueUtils.java

```diff
@@ -429,18 +431,41 @@ private static double findWaterValue(
     return capitalOrFactoryValue / 100 + nearbyLandValue / 10;
   }
 
+  // 1700 oldT=4906 newT=227 newT2=166 newT3=152
+  @SuppressWarnings("ReferenceEquality")
   private static Set<Territory> findNearbyEnemyCapitalsAndFactories(
-      final Territory t, final Map<Territory, Double> enemyCapitalsAndFactoriesMap) {
-
-    Set<Territory> nearbyEnemyCapitalsAndFactories = new HashSet<>();
-    for (int i = MIN_FACTORY_CHECK_DISTANCE; i <= MAX_FACTORY_CHECK_DISTANCE; i++) {
-      nearbyEnemyCapitalsAndFactories = t.getData().getMap().getNeighbors(t, i);
-      nearbyEnemyCapitalsAndFactories.retainAll(enemyCapitalsAndFactoriesMap.keySet());
-      if (!nearbyEnemyCapitalsAndFactories.isEmpty()) {
-        break;
+      final Territory t, final Set<Territory> enemyCapitalsAndFactories) {
+    final GameMap map = t.getData().getMap();
+
+    int distance = 0;
+    Territory nextDistanceAfter = t;
```

Comment:
Sorry to pile on, `nextDistanceAfter` is not very clear to me. I would almost expect the value to be a scalar, similar to distance.  Can we brainstorm some potential better variable names?

- `territoryAtNextDistanceToCheck`?

Reply:
Went with lastTerritoryAtCurrentDistance.

Reply:
Awesome, that is easier, thank you